### PR TITLE
ambient: give ztunnel writable /tmp

### LINKED ISSUE #50060: Ztunnel missing writeable /tmp dir for gathering pprof
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

When attempting to gather a `debug/pprof/profile` for `ztunnel` on a default install the user is met with an empty response
Screenshot 2024-03-21 at 5 14 09 PM

To resolve this, the following `volume` and `volumeMount` is required to add a `/tmp` directory in the ztunnel daemonset manually, or using helm:
```
helm upgrade --install ztunnel istio/ztunnel \
-n kube-system \
--version=1.21.0 \
-f -<<EOF
hub: docker.io/istio
tag: 1.21.0
resources:
  requests:
      cpu: 500m
      memory: 2048Mi
istioNamespace: istio-system
# Additional volumeMounts to the ztunnel container
volumeMounts:
  - name: tmp
    mountPath: /tmp
# Additional volumes to the ztunnel pod
volumes:
  - name: tmp
    emptyDir:
      medium: Memory
EOF
```

Should we be including this missing `volume` and `volumeMount` in the default ztunnel deployment templates? or alternatively, if we decide not to fix the template so it "just works" should we upgrade the handler to have a meaningful reply if `ztunnel` doesn't have the writable location it requires?

cc: @ilrudie @bleggett 

### Version

```Text
% istioctl version
client version: 1.21.0
control plane version: 1.21.0
data plane version: 1.21.0 (25 proxies)

% kubectl version
Client Version: v1.29.0
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.3-gke.1118000
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by bleggett**
**Please provide a description of this PR:**

Fixes https://github.com/istio/istio/issues/50060 - same thing we do in other spots.

### CODE REVIEW

**Comment by zirain** [comment]
need a release notes?

File: manifests/charts/ztunnel/templates/daemonset.yaml [hunk 1]
```diff
@@ -155,6 +157,9 @@ spec:
         hostPath:
           path: /var/run/ztunnel
           type: DirectoryOrCreate # ideally this would be a socket, but ztunnel may not have started yet.
+      # pprof needs a writable /tmp, and we don't have that thanks to `readOnlyRootFilesystem: true`, so mount one
```
Comment (howardjohn) [inline manifests/charts/ztunnel/templates/daemonset.yaml:160]: On envoy we are dumping to `/var/lib/istio/data`. That seems to make sense for consitency vs `/tmp`? And tmp is more likely to be accidentally used by something else... although nothing should be writing to the filesystem, I suppose

Comment (howardjohn) [reply manifests/charts/ztunnel/templates/daemonset.yaml:160]: This would require a change in ztunnel

Comment (bleggett) [reply manifests/charts/ztunnel/templates/daemonset.yaml:160]: > On envoy we are dumping to /var/lib/istio/data. That seems to make sense for consitency vs /tmp? And tmp is more likely to be accidentally used by something else... although nothing should be writing to the filesystem, I suppose

Right - I am inclined to leave it as just `/tmp` - nothing else is running here, the rest of the FS is R/O, and the odds that something else writes on top of a ztunnel pprof file is nil.


We could change ztunnel to make it configurable but in this case it's really not a big deal, IMO.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**